content/en/blog/_posts/2021-10-05-nsa-cisa-hardening.md

```diff
@@ -0,0 +1,413 @@
+in [this task page](https://kubernetes.io/docs/tasks/configure-pod-container/migrate-from-psp/)
+
+One important behavior mentioned in the guidance, that remains the same between
+pod security policy and its replacement is that enforcing either of them does
+not affect _already_ running pods as the enforcement happens at pod creation
+stage.
+
+### Hardening container engines
+
+Some container workloads are less trusted than others but may need to run in the
+same cluster. In those cases, running them on dedicated nodes that include
+hardened container runtimes that provide stricter pod isolation boundaries can
+act as a useful security control.
+
+Kubernetes supports
+a [feature](https://kubernetes.io/docs/concepts/containers/runtime-class/)
+called `RuntimeClass` that is stable / GA (enabled by default) stage as of v1.20
+to natively support this approach. Some projects that support this feature are:
+
+1. [kata containers](https://github.com/kata-containers/kata-containers/blob/main/docs/how-to/how-to-use-k8s-with-cri-containerd-and-kata.md#create-runtime-class-for-kata-containers)
+2. [gvisor](https://gvisor.dev/docs/user_guide/containerd/quick_start/)
+
+As discussed here and in the guidance, many features and tooling in and around
+Kubernetes tooling exists that can enhance the isolation boundaries between two
+pods. Based on relevant threats and risk posture, users should pick and choose
+between them, instead of trying to apply all. Having said that, cluster
+level isolation remains the strictest workload isolation mechanism, inspite of
+improvements mentioned earlier here and in the guide.
+
+## Network Separation and Hardening
+
+Kubernetes Networking can be tricky and this section focuses on how to secure
+and harden the relevant configurations. The guide identifies the following as key
+takeaways: 
+- Using network policies to create isolation between resources, 
+- Securing control plane
+- Encrypting traffic and sensitive data
+
+### Network Policies
+
+Network policies can be created with the help of network plugins. In order to
+make the creation and visualization easier for users, Cilium supports
+a [web GUI tool](https://editor.cilium.io). It creates Kubernetes Network
+policies as well as Cilium Network policies, which can be used to minimize the
+attack vector.
+
+Another scenario that is worth exploring is the usage of external IPs. Some
+services when misconfigured can create random external IPs. An attacker can take
+advantage of this configuration and easily intercept traffic. This vulnerability
+has been reported
+in [CVE-2020-8554](https://www.cvedetails.com/cve/CVE-2020-8554/).
+Using [externalip-webhook]([https://github.com/kubernetes-sigs/externalip-webhook)
+can mitigate this vulnerability. It will prevent the services from using random
+external IPs. [Webhook](https://github.com/kubernetes-sigs/externalip-webhook) 
+will only allow creation of services which doesn't require external IP or whose 
```

Comment:
nit pick: `of services that don't require external IPs or whose external IPs are within the`